### DISCUSSION TITLE
Build Android APK file to provide the huge possibility to attack via ru phone owners.

### DISCUSSION THREAD

**Discussion by oleksiikoshlatyi** [arriven/db1000n#465] [Ideas]
I'm suggesting, this is a possible solution
https://pkg.go.dev/golang.org/x/mobile/cmd/gomobile

**Comment by Wendors** [comment]
This link does not work

**Comment by oleksiikoshlatyi** [comment]
The link is working
![image](https://user-images.githubusercontent.com/32518348/160140091-d5f1c7ef-2685-44a2-86be-15389be16048.jpeg)

**Comment by arriven** [comment]
Seems like an interesting idea, but based on my limited experience with mobile it's likely that the main problem will be crafting a manifest with all the required permissions to use low level attacks, otherwise there might be existing solutions that are better suited for the use case (I've been looking into making a webassembly build and hosting a web version some time ago but it turns out that the browser only allows making http requests from within the sandbox so the benefit is pretty low)

**Comment by oleksiikoshlatyi** [comment]
The requirements (allow an application to use the network) will not be an issue.
APK files can be distributed in many ways, even without a play market (or other applications catalogs).
BTW, almost everyone has an old (2-3 years old) Android phone, that can be used as well as VM rented in public cloud providers.

But with the mobile APP, we have opportunities to use ru phones, in case delivery APK file to mind-full (I still hope, that there are ones who are against the government) people in ru for processing the attacks

**Comment by mimihagi** [comment]
Building and running db1000n in [termux](https://termux.org) on android works without issues. Steps are:

1. install termux
2. install golang, git and make (`apt update && apt install golang git make`)
3. clone and compile db1000n: `git clone https://github.com/arriven/db1000n && cd db1000n && make`
4. Run db1000n: `./db1000n`

> main problem will be crafting a manifest with all the required permissions to use low level attacks

I've run db100n on a phone with termux (as nonroot user) as well as a rpi for the last couple of weeks and have not noticed any permission issues on the phone in the last couple of updates. Was seeing https://github.com/Arriven/db1000n/issues/265 or https://github.com/Arriven/db1000n/issues/200 before, but not anymore

**Comment by 0xDF77-C0R3** [comment]
People familiar with Termux can help even with [Slowloris](https://github.com/gkbrk/slowloris).

`pkg install python`
`pip install slowloris`
`python3 slowloris.py HTTP://URLTOATTACK`

**Comment by zlaya-sobaka** [comment]
Привет, мы с ребятами придумали как можно использовать вашу библиотеку для мобильных устройств Android
https://github.com/zlaya-sobaka/db1000n_mobile
с помощью библиотеки fyne https://github.com/fyne-io/fyne
Сейчас выкатили первую (примитивную) версию которую уже протестировали на мобильных устройствах https://drive.google.com/drive/folders/1boXqWfDEFhtsk6yKJLAULRp8LHee2uLJ?usp=sharing

В приложение добавили logger https://github.com/zlaya-sobaka/db1000n_mobile/blob/main/src/mobilelogger/logger.go
 для мобильных устройств

собирать апк можно библиотекой go имея у себя SDK Android
`fyne release -os android -appID zlaya.sobaka.db1000n_mobile -appVersion 1.0 -appBuild 1 -keyStore ./my-release-key.keystore -icon ./Icon.png`

Возможно это как то поможет выкатить мобильную версия db1000n в мир

**Reply by ghost** [reply]
Are you going to pull-request your's patches to the upstream? I was able to build this app and going to test it soon as well.

**Reply by roman-kruglov** [reply]
@zlaya-sobaka shout out to you, thank you for your attempt. For reference I made a less intrusive (regarding code) attempt inspired by your work here, maybe you'd like to take a look https://github.com/arriven/db1000n/pull/547

**Comment by AdamMickiewich** [comment]
Dzida - a Termux Android App fork that downloads `db1000n` and runs it in background to simplify Android experience.
Because it downloads executable on each start, it will keep up to date `db1000n` inside.

Check it out https://github.com/VolyaTeam/dzida-app/releases/tag/1.0.0

@arriven @arriven FYI
